spring-batch-core/src/main/java/org/springframework/batch/core/configuration/support/AutomaticJobRegistrar.java

```diff
@@ -54,6 +53,10 @@
 
 	private volatile boolean running = false;
 
+	private int phase;
```

Comment:
You should probably default to an early `phase`, such as `Integration.MIN_VALUE + 1000` (leaving room for other components to start earlier, but starting well before components with the default phase (0).

Reply:
Thank you for the review Gary! Indeed, a negative value is better as a default phase. I will update the PR accordingly.

Reply:
I have updated the PR to set the default phase to `Integer.MIN_VALUE + 1000`.